runtime-modules/content/src/benchmarks/benchmarking.rs

```diff
@@ -1797,6 +1792,642 @@ benchmarks! {
+        }
+
+
+    // ================================================================================
+    // ============================ NFT - OFFERS ======================================
+    // ================================================================================
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
+    // - curator owned channel
+    // - curator number is max
+    // - curator has max number of permissions
+    // - channel has all features paused except necessary ones
+    // - channel has max assets
+    // - NFT owner == channel owner
+    // INPUT COMPLEXITY
+    offer_nft {
+        let (channel_id, group_id, lead_account_id, curator_id, curator_account_id) =
+            setup_worst_case_scenario_curator_channel_all_max::<T>(false)?;
+        let origin = RawOrigin::Signed(curator_account_id.clone());
+        let actor = ContentActor::Curator(group_id, curator_id);
+        let video_id = setup_video_with_idle_nft::<T>(curator_account_id.clone(), actor, channel_id)?;
+        let origin = RawOrigin::Signed(curator_account_id);
+        let (_, to_member) = member_funded_account::<T>(DEFAULT_MEMBER_ID);
+        let price = Some(BalanceOf::<T>::one());
+
+    }: _ (origin, video_id, actor, to_member, price)
+        verify {
+            assert!(matches!(Pallet::<T>::video_by_id(video_id).nft_status, Some(Nft::<T> {
+                transactional_status: TransactionalStatus::<T>::InitiatedOfferToMember(to_memeber, price),
+                ..
+            })));
+        }
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
+    // - curator owned channel
+    // - curator number is max
+    // - curator has max number of permissions
+    // - channel has all features paused except necessary ones
+    // - channel has max assets
+    // - NFT owner == channel owner
+    // INPUT COMPLEXITY
+    cancel_offer {
+        let (channel_id, group_id, lead_account_id, curator_id, curator_account_id) =
+            setup_worst_case_scenario_curator_channel_all_max::<T>(false)?;
+        let origin = RawOrigin::Signed(curator_account_id.clone());
+        let actor = ContentActor::Curator(group_id, curator_id);
+        let (_, to_member) = member_funded_account::<T>(DEFAULT_MEMBER_ID);
+        let price = Some(BalanceOf::<T>::one());
+        let video_id = setup_video_with_offered_nft::<T>(curator_account_id.clone(), actor, channel_id, to_member, price)?;
+        let origin = RawOrigin::Signed(curator_account_id);
+
+    }: _ (origin, actor, video_id)
+        verify {
+            assert!(matches!(Pallet::<T>::video_by_id(video_id).nft_status, Some(Nft::<T> {
+                transactional_status: TransactionalStatus::<T>::Idle,
+                ..
+            })));
+        }
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
+    // - curator owned channel
+    // - curator number is max
+    // - curator has max number of permissions
+    // - NFT owner == channel owner
```

Comment:
```suggestion
    // - channel has all features paused
    // - channel has max number of assets
    // - video has max number of assets
    // - nft offer price is_some
    // - nft owner is `Member`
    // - royalty is non-zero
    // - `price - royalty` is non-zero
```

In order for the last 2 conditions to be satisfied, the price cannot be `BalanceOf::<T>::one()`, because then either the royalty or the amount deposited to seller will end up being `0`, which will cause an early return of `deposit_creating` call.

The reason why `NftOwner::Member` is the worst case scenario here, **which is also true for any extrinsic that triggers `complete_payment`**, is that in that case (and assuming royalty is set), the runtime will need to access/modify the data of 3 different accounts: curator channel account (for royalty deposit purposes), nft owner (seller) account (for nft revenue deposit purposes) and the offer "taker" (buyer) account (for usable balance check & slash purposes). If the nft owner is `Channel` then only 2 accounts are affected (channel account and the taker/buyer account)